ydb/core/security/ticket_parser_impl.h

```diff
@@ -804,20 +960,132 @@ class TTicketParserImpl : public TActorBootstrapped<TDerived> {
         SetError(key, record, {.Message = errorMessage, .Retryable = isRetryableError});
     }
 
-    void Handle(NCloud::TEvAccessService::TEvBulkAuthorizeResponse::TPtr& ev) {
-        auto getResourcePathIdForRequiredPermissions = [] (const ::yandex::cloud::priv::accessservice::v2::Resource& resource_path) -> TString {
-            if (resource_path.type() == "resource-manager.folder") {
-                return " folder_id " + resource_path.id();
-            }
-            if (resource_path.type() == "resource-manager.cloud") {
-                return " cloud_id " + resource_path.id();
+    void Handle(NNebiusCloud::TEvAccessService::TEvAuthorizeResponse::TPtr& ev) {
```

Comment:
it looks like a huge copy-pase code? am I wrong? is it possible to avoid or minimize copy of existing code?

Reply:
I've already tried to avoid copy and paste. But what I see is that API is similar, but not the same. For example, there is no UserAccountService/ServiceAccountService in Nebius, similar fields are taken from different places (for example, account field in authorization response is now in each result; results are in map, not in repeated field; there is additional field "authorized" etc). I can create complex templates to make the same code for both APIs, but I think it will be too complex in that case. I looked at the code one more time and refactored several places to that the copy & paste reduced. Look please at the new code and, if you'll see something more to be refactored, I can do it